mlflow/entities/trace.py

```
@@ -100,6 +105,88 @@ def to_pandas_dataframe_row(self) -> dict[str, Any]:
+
+        .. code-block:: python
+
+            import mlflow
+            import re
+            from mlflow.entities import SpanType
+
+
+            @mlflow.trace(span_type=SpanType.CHAIN)
+            def run(x: int) -> int:
+                x = add_one(x)
+                x = add_two(x)
+                x = multiply_by_two(x)
+                return x
+
+
+            @mlflow.trace(span_type=SpanType.TOOL)
+            def add_one(x: int) -> int:
+                return x + 1
+
+
+            @mlflow.trace(span_type=SpanType.TOOL)
+            def add_two(x: int) -> int:
+                return x + 2
+
+
+            @mlflow.trace(span_type=SpanType.TOOL)
+            def multiply_by_two(x: int) -> int:
+                return x * 2
+
+
+            # Run the function and get the trace
+            y = run(2)
+            trace = mlflow.get_last_active_trace()
+
+            # 1. Search spans by name (exact match)
+            spans = trace.search_spans(name="add_one")
+            print(spans)
+            # Output: [Span(name='add_one', ...)]
+
+            # 2. Search spans by name (regular expression)
+            pattern = re.compile("add.*")
+            spans = trace.search_spans(name=pattern)
+            print(spans)
+            # Output: [Span(name='add_one', ...), Span(name='add_two', ...)]
+
+            # 3. Search spans by type
+            spans = trace.search_spans(span_type=SpanType.LLM)
+            print(spans)
+            # Output: [Span(name='run', ...)]
+
+            # 4. Search spans by name and type
+            spans = trace.search_spans(name="add_one", span_type=SpanType.TOOL)
+            print(spans)
+            # Output: [Span(name='add_one', ...)]
+        """
+
+        def _match_name(span: Span) -> bool:
+            if isinstance(name, str):
+                return span.name == name
+            elif isinstance(name, re.Pattern):
+                return name.match(span.name) is not None
+            return True
```

What shoud if a user passes an int to name by mistake?

---

Good callout, we should raise an error for such case.